src/components/myCart/MyCartItem.tsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import {Image, Platform, Text, TouchableOpacity, View} from 'react-native';
+import {styles} from './styles';
+import ProductAttributes from '../../utils/ProductAttributes';
+import {Icon} from 'react-native-elements';
+import {COLORS} from '../../utils/colors';
+
+type MyCartItemProp = {
+  name: string;
+  title: string;
+  color: string;
+  total: string;
+  discount: string;
+  currency: string;
+  itemCount: number;
+  onDelete: any;
+  onPlus: any;
+  onMinus: any;
+  cart: {};
+};
+
+const MyCartItem = ({
+  name,
+  title,
+  color,
+  total,
+  discount,
+  currency,
+  itemCount,
+  onDelete,
+  onPlus,
+  onMinus,
+  cart,
+}: MyCartItemProp) => {
+  const PLUS_PNG_PATH: string = '../../assets/plus.png';
+  const MINUS_PNG_PATH: string = '../../assets/minus.png';
+
+  if (title && total && discount && currency) {
+    let imageUri: string;
+    let lineItem: number;
+    let quantity: number;
+    for (let i = 0; i < cart.included.length; i++) {
+      if (
+        title &&
+        color &&
+        total &&
+        discount &&
+        currency &&
+        cart.included[i].id === name
+      ) {
+        const variantId = cart.included[i].relationships.variant.data.id;
+        quantity = cart.included[i].attributes.quantity;
+        for (let j = 0; j < cart.included.length; j++) {
+          if (
+            cart.included[j].id === variantId &&
+            cart.included[j].type === 'variant'
+          ) {
+            const imageId = cart.included[j].relationships.images.data[0].id;
+            for (let k = 0; k < cart.included.length; k++) {
+              if (
+                cart.included[k].id === imageId &&
+                cart.included[k].type === 'image'
+              ) {
+                imageUri = cart.included[k].attributes.original_url;
+                lineItem = cart.data.relationships.line_items.data[i].id;
+              }
+            }
+          }
+        }
+      }
+    }
```

Comment:
Such complex logic blocs should be moved out of the component.
It's pretty convenient to keep your component as "dumb" as possible. It makes it easier to sustain more clear codebase.

Also, this function by itself has a lot to improve, ideally, it should be refactored.